libsolidity/ast/Types.h

```diff
@@ -338,9 +384,13 @@ class IntegerConstantType: public Type
 
 	/// @returns the smallest integer type that can hold the value or an empty pointer if not possible.
 	std::shared_ptr<IntegerType const> integerType() const;
-
+	/// @returns the smallest fixed type that can hold the value or an empty pointer
+	std::shared_ptr<FixedPointType const> fixedPointType() const;
+	bigint denominator() const { return m_value.denominator(); }
+	
 private:
-	bigint m_value;
+	rational m_value;
+	short m_scalingFactor;
```

Comment:
yes but remember, the NumberConstant and especially the fixedPoint function don't necessarily know that without an input.